### DISCUSSION TITLE
Welcome to CustomSupportCylinder Discussions!

### DISCUSSION THREAD

**Discussion by 5axes** [5axes/CustomSupportCylinder#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pierremilcent** [comment]
Hello,
i am facing a specific demand, that some others have also. 

I am printing parts for plane in 3D, fuselage & wings. Wings are the most difficult pieces.
When printing very narrow parts with some height, more than  8 to 10cm, i have a wobble effect on the most of the top which increases a lot at the higher part.
I can reduce this effect partially with a better orientation, either in the diagonal of the bed of in the Y axis, while the head is printing/translating  mainly on this axe.
Also if i reduce the speed this is improving too.

But at the end the shaking effect is still there. You can see this and some improvement on the picture below.

The only solution would be to have some support to fix the higher part of the piece and avoid any lateral movement.
Someone, and i also tried, built some specific parts with supports like this one, but it needs to go into Fusion with the STL and add manually these supports.

Regarding your expertise in this domain, and your Cura extensions are very valuable and efficient, i wonder if you could add such kind of support in your development.

Letr us know what do you think.
Don't worry about the stringing since i am printing with LW-PLA this material has always stringing which can easily be removed.

**Part sample.**

![image](https://user-images.githubusercontent.com/38214885/110383174-376a6c00-805c-11eb-8e98-8932e2690e33.png)

**Before improvement**

![IMG_2566](https://user-images.githubusercontent.com/38214885/110381515-d772c600-8059-11eb-9835-478a1864298a.jpg)

**After Improvement, orientation + speed**

![IMG_2567](https://user-images.githubusercontent.com/38214885/110381513-d6419900-8059-11eb-9fcf-6a7fef3488f8.jpg)

![IMG_2568](https://user-images.githubusercontent.com/38214885/110381511-d5106c00-8059-11eb-9207-125d48007f45.jpg)

Example of support we could have.

![155993895_4107648595945839_7842997772606415769_o](https://user-images.githubusercontent.com/38214885/110381604-fd986600-8059-11eb-95fb-f2fe43bbe2e8.jpg)

**Comment by 5axes** [comment]
In the next release 2.5.0 you will have the possibility to use a FreeFrom Support.  And a Custom support will give you something not similar to your request. Not exactly the same shape but something close in the idea.
![image](https://user-images.githubusercontent.com/11015345/110387868-8adfb880-8062-11eb-812f-9799fa4a3d4d.png)

**Reply by pierremilcent** [reply]
I just found this early version before your answer and i am testing it.
I just have the default shape, no custom option as on your screenshot.
Will let you know how it goes.

![2021-03-08 22_56_41-Greenshot](https://user-images.githubusercontent.com/38214885/110391024-2f63f980-8067-11eb-8ba8-2dbcf7d6ddca.png)

![2021-03-08 22_59_05-Greenshot](https://user-images.githubusercontent.com/38214885/110391029-31c65380-8067-11eb-9a19-54978c28d321.png)

**Reply by 5axes** [reply]
Just because you download it too early :)

**Reply by pierremilcent** [reply]
Arg ... will try again later so ... Thanks.

**Reply by pierremilcent** [reply]
Early result is very promising since the surface of the print is now much more smooth.
I need to manage how many support and not too close of the part.

Waiting for the final release with optional support form may be ? Thanks.

![IMG_2569](https://user-images.githubusercontent.com/38214885/110447293-5ef32000-80c0-11eb-8839-75852104c4fc.jpg)

![IMG_2570](https://user-images.githubusercontent.com/38214885/110447306-61ee1080-80c0-11eb-8ee0-329c7abcb35d.jpg)

![IMG_2571](https://user-images.githubusercontent.com/38214885/110447312-64506a80-80c0-11eb-8f3b-415757c675c6.jpg)

![IMG_2572](https://user-images.githubusercontent.com/38214885/110447326-661a2e00-80c0-11eb-89b5-c5a5984f0cca.jpg)

**Reply by 5axes** [reply]
The version is available since last night. I won't call it a "final" version but more the beginning of an evolution. The idea is to allow users to define their own support.  The principle is to use some STL files that could be modified, the base being created from OpenScad models and to have no limit in the number of support geometry (As I don't have the possibility to get the normal direction we could create a X+ Y+ X- Y- support to define an initial orientation). It will be up to each user to create either an efficient support or, why not, if the feedback is important, to create a base of efficient support fior different application  (but I have little confidence on this point, as users are often more first level users than actors).

**Comment by 5axes** [comment]
This is really not a final : in a few minutes a 2.5.1 with option to rotate the FreeForm Support.
![image](https://user-images.githubusercontent.com/11015345/110467256-17c45980-80d7-11eb-816c-aafd59038b8c.png)
Looks ok for your application (select  Y direction option or mirror ( it's not really mirror but a 180° rotation) to place de support
![image](https://user-images.githubusercontent.com/11015345/110469112-84d8ee80-80d9-11eb-9069-f57a5a6d27c9.png)

**Reply by pierremilcent** [reply]
Nice !! will test it soon

**Reply by 5axes** [reply]
Release 2.5.2 (With Bridge type) A short video concerning these new options :  [https://www.youtube.com/watch?v=lCSbLjEbc-E](https://www.youtube.com/watch?v=lCSbLjEbc-E)

**Reply by pierremilcent** [reply]
Very good result, do no worry about stringing its LW-PLA, which does stringing a lot, retraction almost impossible with this one.

![IMG_2602](https://user-images.githubusercontent.com/38214885/110542925-f476ca00-8129-11eb-8106-a6b9aab35537.jpg)

![IMG_2598](https://user-images.githubusercontent.com/38214885/110542931-f6408d80-8129-11eb-85da-6c1327148c54.jpg)

**Before / After**

![IMG_2578](https://user-images.githubusercontent.com/38214885/110543090-2851ef80-812a-11eb-9031-c457564dd747.jpg)

![IMG_2576](https://user-images.githubusercontent.com/38214885/110543096-2a1bb300-812a-11eb-89e7-596bcafa3c3e.jpg)

**Reply by pierremilcent** [reply]
> Release 2.5.2 (With Bridge type) A short video concerning these new options : https://www.youtube.com/watch?v=lCSbLjEbc-E

Is there any way to know the plugin / extension version that is installed in Cura ?? If not it could be a suggestion for next Cura vrsion.

**Reply by 5axes** [reply]
Yes Marketplace -> Installed
![image](https://user-images.githubusercontent.com/11015345/110547532-59352300-8130-11eb-9f37-d42089959279.png)

Par contre la question c'est pourquoi on échange en Anglais  ?

**Reply by pierremilcent** [reply]
Parce que cela peut aussi aider plus de monde que le simple cercle de ceux parlant français. Mais maintenant que la 2.5.2 est disponible on peut parler uniquement français.
Merci cette extension donc, qui sauve mes impressions en LW-PLA d'avions ... Les ailes sont fines et donc ça bouge pas mal.
[https://youtu.be/9ap7rGn3uh8](url)
[https://youtu.be/i-0uRGjPt74](url)

**Reply by pierremilcent** [reply]
Y a t il une raison pour avoir cette différence de version affichée sur le marketplace ?

![2021-03-10 01_15_51-Greenshot](https://user-images.githubusercontent.com/38214885/110556258-6b1dc280-813e-11eb-8df3-0659cdfa849f.png)

![2021-03-10 01_15_42-Greenshot](https://user-images.githubusercontent.com/38214885/110556275-783ab180-813e-11eb-813e-660a1e1b7cba.png)

**Reply by 5axes** [reply]
Oui, la version 2.5.2 c'est celle que tu as installé à partir du Github et celle sur mon Pc. L'autre celle disponible actuellement sur la marketplace et ca ca ne dépend pas que de moi mais des équipes Ultimaker qui décide ou non de valider un Plugin . En général si tout va bien compter 7 à 10 jours avant d'avoir une version disponible via le market place ( si le plugin est validé)

**Reply by pierremilcent** [reply]
Justement j'ai désinstallé la version provenant de Git, pour avoir les MAJ en provenance de la market place. Juste un décalage dans le temps sans doute dans la plate forme de Utimaker.

**Comment by 5axes** [comment]
Ok working not so bat, support really easy to remove with a x/y distance of 0.18
![20210310_074903](https://user-images.githubusercontent.com/11015345/110591266-25cab680-8179-11eb-9895-f933b6a9e852.jpg)

![20210310_074622](https://user-images.githubusercontent.com/11015345/110591198-0cc20580-8179-11eb-98e7-6dd994b913ed.jpg)

**Comment by 5axes** [comment]
The 2.5.3 Release will offer this support (minor modification to the initial shape according to my test) as an "arch buttress" Support "arc boutant"
![image](https://user-images.githubusercontent.com/11015345/110602015-d8087b00-8185-11eb-96a1-55974282ac17.png)

**Reply by pierremilcent** [reply]
Nice !

**Comment by pierremilcent** [comment]
Petite interrogation avec le support "custom" : voici une pièce que j'ai imprimé sans les supports.
Il y a à la base une  patte qui sert de d'attache pour une commande de servo (c'est un morceau d'aile d'avion).

![2021-03-10 17_46_35-Papa’s Best STL Viewer – eagle-Aileron1_LW-left-profile3 stl](https://user-images.githubusercontent.com/38214885/110668171-4d4a6f00-81cb-11eb-8b92-d6f652326a62.png)

Sous cura son épaisseur est de 6 lignes (en 0.25)

![2021-03-10 17_47_37-E3PROSKR_eagle-Aileron1_LW-left-profile3_PPPLWPLA-1-27-025 - Ultimaker Cura 4 8](https://user-images.githubusercontent.com/38214885/110668350-81be2b00-81cb-11eb-82ab-71e198c952e5.png)
![2021-03-10 18_06_55-E3PROSKR_eagle-Aileron1_LW-left-profile3_PPPLWPLA-1-27-025 - Ultimaker Cura 4 8](https://user-images.githubusercontent.com/38214885/110668356-8387ee80-81cb-11eb-94b5-a0bdfc3e41dd.png)


Par contre dès que je pose des supports l'impression de cette patte est "escamottée" seul le pourtour des 6 lignes est réalisé comme on voit ci dessous.
Je pense que c'est lié à un paramètre par défaut dans les supports qui sont pris en compte dès l'ajout d'un support d'une extension, même si la case support n'est pas cochée (ce qui doit être le cas pour les supports spéciaux).
Mais du coup impossible de trouver quel paramètre influe sur cette impression ou absence de contenu dans cette partie.
Une idée ?

![2021-03-10 17_47_09-E3PROSKR_eagle-Aileron1_LW-left-profile3_PPPLWPLA-2-08-025 - Ultimaker Cura 4 8](https://user-images.githubusercontent.com/38214885/110668495-a914f800-81cb-11eb-8c21-6150982fb2aa.png)

![2021-03-10 18_09_33-E3PROSKR_eagle-Aileron1_LW-left-profile3_PPPLWPLA-1-33-025 - Ultimaker Cura 4 8](https://user-images.githubusercontent.com/38214885/110668638-cea20180-81cb-11eb-91f3-88eb766823ce.png)

**Reply by pierremilcent** [reply]
Enfin si j'active les supports standards la partie basse de cette pièce est bien prise en compte.
Donc le souci provient du paramétrage du support custom je pense.

![2021-03-10 18_40_17-E3PROSKR_eagle-Aileron1_LW-left-profile3_PPPLWPLA-1-38-025 - Ultimaker Cura 4 8](https://user-images.githubusercontent.com/38214885/110672834-5b4ebe80-81d0-11eb-992d-69fe08fbdca7.png)

**Reply by 5axes** [reply]
PAs certains que ce soit lié au support ? à la base la pièce est bizarre que des contours rouge ( donc externe) devrait y avoir du vert au moins ..  A mon avois ca vient de tes réglages .. mais comme ça pas facile à dire.

**Reply by 5axes** [reply]
Note 2 ... Théoriquement avec les supports custom on active pas les supports standards

**Reply by pierremilcent** [reply]
OK. C'est un profil spécifique en effet pour imprimer avec une paroi. Je vais essayer avec un profil de base Cura 0.2

**Reply by pierremilcent** [reply]
> Note 2 ... Théoriquement avec les supports custom on active pas les supports standards

Mais ce sont les valeurs des supports qui sont prises en compte néanmoins, n'est ce pas ?

**Reply by 5axes** [reply]
Oui

**Comment by lixiaodao1234** [comment]
Can achieve with the mouse frame out of the selection and then generate support

**Comment by lixiaodao1234** [comment]
Can achieve with the mouse frame out of the selection and then generate support
